Detectors/MUON/MCH/Raw/Decoder/src/UserLogicEndpointDecoder.h

```diff
@@ -171,19 +167,18 @@ size_t UserLogicEndpointDecoder<CHARGESUM>::append(Payload buffer)
       }
     }
 
-    // bits 50..52 are error bits
-    int8_t error = static_cast<uint8_t>((word >> 50) & 0x7);
+    int8_t error = ulword.error;
+    int8_t incomplete = ulword.incomplete;
+    uint64_t data50 = ulword.data;
 
-    // the remaining (LSB) 50 bits represents the actual data to passed to the ElinkDecoder
-    uint64_t data50 = word & FIFTYBITSATONE;
-    d->second.at(dsid).append(data50, error);
+    d->second.at(dsid).append(data50, error | (incomplete << 2));
```

Comment:
In the V1 data format we have three bits reserved to the error code, while the code above puts the incomplete bit at bit number 2, leaving only two bits for the error code.

For better flexibility, wouldn't it be better to pass the incomplete flag as a third parameter of the `append()` function?

Reply:
indeed, my treatment of error + incomplete is wrong here. Will change that.

Reply:
@aferrero2707 actually one question : can I safely consider that incomplete will only ever be 1 bit = boolean ?

Reply:
Yes, I think so. We are in any case quite limited in the number of control bits for the UL data words, so I doubt we will ever use more than 1 bit for the incomplete word condition.